File: sample-java/src/main/java/fr/bmartel/bboxapi/javasample/GetApps.java
package fr.bmartel.bboxapi.javasample;

import com.github.kittinunf.fuel.core.*;
import com.github.kittinunf.result.Result;
import de.mannodermaus.rxbonjour.platforms.desktop.DesktopPlatform;
import fr.bmartel.bboxapi.stb.BboxApiStb;
import fr.bmartel.bboxapi.stb.model.Application;
import kotlin.Triple;
import kotlin.Unit;

import java.util.List;

public class GetApps {

    public static void main(String args[]) {
        BboxApiStb bboxapi = new BboxApiStb("YourAppId", "YourAppSecret");

        bboxapi.startRestDiscovery(true, DesktopPlatform.create(), 10000, (stbServiceEvent, stbService, throwable) -> {
            switch (stbServiceEvent) {
                case SERVICE_FOUND:
                    System.out.println("service found : " + stbService.getIp() + ":" + stbService.getPort());

                    bboxapi.getApps(new Handler<List<Application>>() {
                        @Override
                        public void failure(Request request, Response response, FuelError error) {
                            if (error.getException() instanceof HttpException) {
                                System.out.println("http error : " + response.getStatusCode());
                            } else {
                                error.printStackTrace();
                            }
                        }

                        @Override
                        public void success(Request request, Response response, List<Application> data) {
                            System.out.println(data);
                        }
                    });

                    Triple<Request, Response, Result<List<Application>, FuelError>> data = bboxapi.getAppsSync();
                    Request request = data.getFirst();
                    Response response = data.getSecond();
                    Result<List<Application>, FuelError> obj = data.getThird();
                    System.out.println(obj.get());
                    break;
                case DISCOVERY_STOPPED:
                    System.out.println("end of discovery");
                    break;
                case DISCOVERY_ERROR:
                    throwable.printStackTrace();
                    break;
            }
            return Unit.INSTANCE;
        });
    }
}

File: sample/src/main/kotlin/fr/bmartel/bboxapi/stb/sample/GetApps.kt
package fr.bmartel.bboxapi.stb.sample

import com.github.kittinunf.fuel.core.HttpException
import com.github.kittinunf.result.Result
import de.mannodermaus.rxbonjour.platforms.desktop.DesktopPlatform
import fr.bmartel.bboxapi.stb.BboxApiStb
import fr.bmartel.bboxapi.stb.model.StbServiceEvent

fun main(args: Array<String>) {
    val bboxapi = BboxApiStb(appId = "YourAppId", appSecret = "YourAppSecret")

    bboxapi.startRestDiscovery(findOneAndExit = true, maxDuration = 10000, platform = DesktopPlatform.create()) { eventType, service, changed, error ->
        when (eventType) {
            StbServiceEvent.SERVICE_FOUND -> {
                bboxapi.getApps { _, response, result ->
                    when (result) {
                        is Result.Failure -> {
                            val ex = result.getException()
                            when {
                                ex.exception is HttpException -> println("http error : ${response.statusCode}")
                                else -> ex.printStackTrace()
                            }
                        }
                        is Result.Success -> {
                            val data = result.get()
                            println(data)
                        }
                    }
                }

                val (_, _, result) = bboxapi.getAppsSync()
                when (result) {
                    is Result.Failure -> {
                        result.getException().printStackTrace()
                    }
                    is Result.Success -> {
                        println(result.get())
                    }
                }
            }
            StbServiceEvent.DISCOVERY_STOPPED -> println("end of discovery")
            StbServiceEvent.DISCOVERY_ERROR -> error?.printStackTrace()
        }
    }
}